hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/clientrm/TestFederationClientInterceptorRetry.java

```diff
@@ -283,4 +300,58 @@ public void testSubmitApplicationOneBadOneGood()
     SubClusterId respSubClusterId = responseHomeSubCluster.getHomeSubCluster();
     Assert.assertEquals(good, respSubClusterId);
   }
+
+  @Test
+  public void testSubmitApplicationTwoBadOneGood() throws Exception {
+
+    LOG.info("Test submitApplication with two bad, one good SC.");
+
+    // This test must require the TestSequentialRouterPolicy policy
+    if (StringUtils.equals(routerPolicyManagerName,
```

Comment:
It may make sense to use something like:
```
Assume.assumeThat()
```

Reply:
Thanks for your suggestion, I will modify the code.